src/Sarif.Viewer.VisualStudio.Core/Telemetry/KeyEventTelemetry.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+
+using Microsoft.Sarif.Viewer.Shell;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.Sarif.Viewer.Telemetry
+{
+    internal class KeyEventTelemetry
+    {
+        internal const string Product = "VS/Extensions/SarifViewer/KeyEvents/";
+
+        private static string vsVersion;
+        private static string extensionVersion;
+
+        private readonly ITelemetryClient telemetryClient;
+
+        public static string VsVersion
+        {
+            get
+            {
+                if (SarifViewerPackage.IsUnitTesting)
+                {
+                    return string.Empty;
+                }
+
+                vsVersion ??= VsUtilities.GetVsVersion() ?? string.Empty;
+                return vsVersion;
+            }
+        }
+
+        public static string ExtensionVersion
+        {
+            get
+            {
+                extensionVersion ??= Assembly.GetExecutingAssembly().GetName().Version.ToString();
+                return extensionVersion;
+            }
+        }
+
+        public KeyEventTelemetry(ITelemetryClient telemetryClient = null)
+        {
+            this.telemetryClient = telemetryClient ?? new KeyEventTelemetryClient();
+        }
+
+        public void TrackEvent(string eventName, TelemetryResult result = TelemetryResult.Success)
+        {
+            if (eventName == null)
+            {
+                throw new ArgumentNullException(nameof(eventName));
+            }
+
+            UserTaskEvent trackEvent = CreateEvent(eventName, result);
+
+            this.telemetryClient.PostEvent(trackEvent);
+        }
+
+        public void TrackEvent(string eventName, Dictionary<string, string> properties, TelemetryResult result = TelemetryResult.Success)
```

Comment:
>Dictionary<string, string> properties, [](http://example.com/codeflow?start=49&length=38)

Make this the third parameter

Reply:
combined 2 functions and make the properties 3rd parameter.